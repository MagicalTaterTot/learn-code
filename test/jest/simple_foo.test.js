import React from 'react';
import { shallow, mount, render } from 'enzyme';
import toJson from 'enzyme-to-json';

import SimpleFoo from '../../src/components/simple_foo';

describe('A suite', function() {
  it('should render', () => {
    const wrapper = shallow(<SimpleFoo />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
    
  it('should render without throwing an error', function() {
    expect(shallow(<SimpleFoo />).contains(<div className="foo">Simple Bar</div>)).toBe(true);
  });

  it('should be selectable by class "foo"', function() {
    expect(shallow(<SimpleFoo />).is('.foo')).toBe(true);
  });

  it('should mount in a full DOM', function() {
    expect(mount(<SimpleFoo />).find('.foo').length).toBe(1);
  });

  it('should render to static HTML', function() {
    expect(render(<SimpleFoo />).text()).toEqual('Simple Bar');
  });
});
