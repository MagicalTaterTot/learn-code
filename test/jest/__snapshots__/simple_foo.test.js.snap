// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A suite should render 1`] = `
<div
  className="foo"
>
  Simple Bar
</div>
`;
